tests/api_connexion/endpoints/test_pool_endpoint.py

```diff
@@ -141,3 +141,164 @@ def test_response_404(self):
+        self.assertEqual(
+            {
+                "name": "test_pool_a",
+                "slots": 3,
+                "occupied_slots": 0,
+                "running_slots": 0,
+                "queued_slots": 0,
+                "open_slots": 3,
+            },
+            response.json,
+        )
+
+    @provide_session
+    def test_response_409(self, session):
+        pool_name = "test_pool_a"
+        pool_instance = Pool(pool=pool_name, slots=3)
+        session.add(pool_instance)
+        session.commit()
+        response = self.client.post(
+            "api/v1/pools", json={"name": "test_pool_a", "slots": 3}
+        )
+        assert response.status_code == 409
+        self.assertEqual(
+            {
+                "detail": None,
+                "status": 409,
+                "title": f"Pool: {pool_name} already exists",
+                "type": "about:blank",
+            },
+            response.json,
+        )
+
+    @parameterized.expand(
+        [
+            ("for missing pool name", {"slots": 3}, "name"),
+            ("for missing slots", {"name": "invalid_pool"}, "slots"),
+        ]
+    )
+    def test_response_400(self, name, request_json, missing_field):  # pylint: disable=unused-argument
+        response = self.client.post("api/v1/pools", json=request_json)
+        assert response.status_code == 400
+        self.assertEqual(
+            {
+                "detail": f"'{missing_field}' is a required property",
+                "status": 400,
+                "title": "Bad Request",
+                "type": "about:blank",
+            },
+            response.json,
+        )
+
+
+class TestPatchPool(TestBasePoolEndpoints):
+    @provide_session
+    def test_response_200(self, session):
+        pool = Pool(pool="test_pool", slots=3)
+        session.add(pool)
+        session.commit()
+        response = self.client.patch(
+            "api/v1/pools/test_pool", json={"name": "test_pool_a", "slots": 3}
+        )
+
+        assert response.status_code == 200
+        self.assertEqual(
+            {
+                "occupied_slots": 0,
+                "queued_slots": 0,
+                "name": "test_pool_a",
+                "open_slots": 3,
+                "running_slots": 0,
+                "slots": 3,
+            },
+            response.json,
+        )
+
+    def test_response_404(self):
+        response = self.client.patch(
+            "api/v1/pools/test_pool", json={"name": "test_pool_a", "slots": 3}
+        )
+        assert response.status_code == 404
+        self.assertEqual(
+            {
+                "detail": None,
+                "status": 404,
+                "title": "Pool not found",
+                "type": "about:blank",
+            },
+            response.json,
+        )
+
+
+class TestPatchPoolWithUpdateMask(TestBasePoolEndpoints):
+    @parameterized.expand(
+        [
+            (
+                "api/v1/pools/test_pool?update_mask=name, slots",
```

Comment:
Can you add a test that tries to update the read-only field e.g. occupied_slots?

Reply:
Also a test where there's a typo in the fields in update mask?

Reply:
> Can you add a test that tries to update the read-only field e.g. occupied_slots?

Here is the [change](https://github.com/apache/airflow/pull/9329/files#diff-83587b0e0b4bb59a06b380179eb8e914R318-R329)

Reply:
> Also a test where there's a typo in the fields in update mask?

Here is [change](https://github.com/apache/airflow/pull/9329/files#diff-83587b0e0b4bb59a06b380179eb8e914R330-R341)